### ISSUE TITLE
shell created by popen cannot find gnuplot

### ISSUE THREAD

**Issue by madronalabs** [alandefreitas/matplotplusplus#94]
**Bug category**
- [ ] *bug - compilation error*
- [ ] *bug - compilation warning*
- [X] *bug - runtime error*
- [ ] *bug - runtime warning*
- [ ] *bug - logic error*

**Describe the bug**
Running my first example failed because the PATH variable of the shell made by popen did not allow it to find gnuplot.

**Steps to Reproduce**

- install matplotplusplus from the package provided by Homebrew
- copy the example program plot_1.cpp into a new project
- compile the project, linking to libmatplot.a

**Output**

<details>
 
```console
sh: command not found: gnuplot
```
</details>

**Platform**

- [ ] *cross-platform issue - linux*
- [ ] *cross-platform issue - windows*
- [X] *cross-platform issue - macos*

**Environment Details:**
 - OS: 10.15.6 (Catalina)
 - OS Version: 
 - Compiler: Apple Clang
 - Compiler version: 11.0.3

**Additional context**

I don't know whether this is something that broke with Catalina. I tried making sh launch bash as well as zsh and they both have this issue. In particular, zsh is required to load  /etc/zshenv on startup whether the shell is interactive or not, however it is clearly not doing this when launched by popen, with the sad result that it can't find commands in /usr/local/bin.

While the Mac OS behavior leading to this issue is clearly broken, there should be a way to address this without having to build matplotplusplus from source. I did go down that path, and ran into other problems. 

The command "gnuplot" is hard-coded into the popen command in gnuplot.cpp. Adding an option in the matplot++ API to set this to a different command (in my case "usr/local/bin/gnuplot") or add a path would be one workaround for the problem I am facing here.

**Comment by alandefreitas** [comment]
I'm not sure I understand. You have `usr/local/bin` in your `PATH` variable but `popen` creates a new shell where `PATH` is not available? I've never seen this behavior anywhere. I'm on Catalina 10.15.6 right now, actually.

In any case, we could pick the GNUPLOT command from an environment variable, but that would probably be at least as much effort as setting the PATH variable. 

It's very easy to create a command to define which command to use in the backend. No problem here. I'll add it to the list. 

I'm not sure I would personally use it in a library though. Because the command still has to come from somewhere: In principle, this allows people to create software that scans the system looking for gnuplot in runtime, etc, etc. In practice, doing this is not really worth the effort, and the only use case I can think of is hard-coding very specific paths in their code.

**Comment by madronalabs** [comment]
> You have usr/local/bin in your PATH variable but popen creates a new shell where PATH is not available? I've never seen this behavior anywhere. I'm on Catalina 10.15.6 right now, actually.

Yes that's correct. But since you are on Catalina and not seeing this issue, it seems likely there is something wrong with my particular setup. If I can figure anything else out, I'll follow up here.

**Comment by alandefreitas** [comment]
OK. Just in case, I'll also create a new function in the gnuplot backend API for choosing a custom command.

**Comment by madronalabs** [comment]
👍 thanks!

**Comment by warycat** [comment]
I have gnuplot installed. I get the same error. I am building a hello world on apple m2 macbook air.
```
(base) ➜  ~ which gnuplot
/opt/homebrew/bin/gnuplot
```
```
sh: gnuplot: command not found
```

**Comment by ItsTeccs** [comment]
> OK. Just in case, I'll also create a new function in the gnuplot backend API for choosing a custom command.

Was this ever addressed and added? I have a use-case as well where I cannot add the path to gnuplot to the entire system environment before running. A configuration to choose a custom command so I can target the custom directory where gnuplot is located would be helpful.
